### PR TITLE
Add remaining cluster-user commands

### LINKED ISSUE #1221: Add `neuro admin add-cluster-user`
```
neuro admin add-cluster-user <cluster-name> <user-name> [role: admin|manager|user]
```

`user` is the default role - could be omitted.

```
curl -vv -H "Authorization: Bearer $TOKEN" -X POST -d '{"user_name": "user-name", "role": "user"}' https://dev.neu.ro/apis/admin/v1/clusters/$CNAME/users
```

https://docs.google.com/document/d/1A9STpF-9vlaVAaCisCqo-o7NU-Kikc9aFsEHOwImuKY/edit#

### LINKED ISSUE #1222: Add `neuro admin remove-cluster-user`
```
neuro admin remove-cluster-user <cluster-name> <user-name>
```

```
curl -vv -H "Authorization: Bearer $TOKEN" -X DELETE https://dev.neu.ro/apis/admin/v1/clusters/$CNAME/users/$UNAME

```

### PR DESCRIPTION
**Pull request by tvoinarovskyi**
closes #1222 
closes #1221

### CODE REVIEW

File: neuromation/cli/admin.py [hunk 1]
```diff
@@ -26,4 +26,31 @@ def admin() -> None:
...
+
+@command()
+@click.argument("cluster_name", required=True, type=str)
+@click.argument("user_name", required=True, type=str)
+@async_cmd()
+async def remove_cluster_user(root: Root, cluster_name: str, user_name: str) -> None:
+    """
+    Remove user access from the cluster
+    """
+    await root.client._admin.remove_cluster_user(cluster_name, user_name)
```
Comment (tvoinarovskyi) [inline neuromation/cli/admin.py]: Should we print something to the user, like: 
Added user: Ivan[admin]
Removed user: Alex

Comment (asvetlov) [reply neuromation/cli/admin.py]: I think yes.
Ideally, we should print texts as you had in examples in the normal mode and skip printing if `--quiet` is specified.
Adding ANSI escapes (bold, underline, etc.) can improve visual readability, see `neuro config show` and others for inspiration.

File: neuromation/api/admin.py [hunk 2]
```diff
@@ -38,6 +38,26 @@ def __init__(self, core: _Core, config: Config) -> None:
...
+        self, cluster_name: str, user_name: str, role: str
+    ) -> _ClusterUser:
+        role_type = _ClusterUserRoleType(role)
+        user = _ClusterUser(user_name, role_type)
+        url = self._config.admin_url / "clusters" / cluster_name / "users"
+        payload = {"user_name": user_name, "role": role}
+        auth = await self._config._api_auth()
+
+        async with self._core.request("POST", url, json=payload, auth=auth) as resp:
+            await resp.json()
```
Comment (serhiy-storchaka) [inline neuromation/api/admin.py]: Loading and parsing a JSON is not needed if the value is ignored.

Or maybe use `_cluster_user_from_api()` for creating the returning value?

File: neuromation/cli/admin.py [hunk 3]
```diff
@@ -26,4 +26,31 @@ def admin() -> None:
     )
 
 
+@command()
+@click.argument("cluster_name", required=True, type=str)
+@click.argument("user_name", required=True, type=str)
+@click.argument("role", required=False, default="user", type=str)
```
Comment (serhiy-storchaka) [inline neuromation/cli/admin.py]: Maybe use `click.Choice`?

**Comment by serhiy-storchaka** [comment]
Maybe add tests for errors? Removing non-existing user, adding already existing user?

File: neuromation/api/admin.py (hunk 2 above)
Comment (tvoinarovskyi) [reply neuromation/api/admin.py]: Probably will do the 2nd one.

File: neuromation/cli/admin.py [hunk 4]
```diff
@@ -38,6 +38,11 @@ def admin() -> None:
     Add user access to specified cluster with one of 3 roles: admin, manager or user
     """
     await root.client._admin.add_cluster_user(cluster_name, user_name, role)
+    if root.verbosity >= 0:
```
Comment (serhiy-storchaka) [inline neuromation/cli/admin.py]: Or `not root.quiet`.

File: neuromation/cli/admin.py (hunk 1 above)
Comment (tvoinarovskyi) [reply neuromation/cli/admin.py]: Added echo calls with quiet support

File: neuromation/cli/admin.py (hunk 3 above)
Comment (tvoinarovskyi) [reply neuromation/cli/admin.py]: done

File: neuromation/cli/admin.py (hunk 4 above)
Comment (tvoinarovskyi) [reply neuromation/cli/admin.py]: done

File: neuromation/cli/admin.py [hunk 5]
```diff
@@ -26,4 +28,46 @@ def admin() -> None:
     )
 
 
+@command()
+@click.argument("cluster_name", required=True, type=str)
+@click.argument("user_name", required=True, type=str)
+@click.argument(
+    "role",
+    required=False,
+    default="user",
```
Comment (dalazx) [inline neuromation/cli/admin.py]:
```suggestion
    default= _ClusterUserRoleType.USER.value,
```
minor.

**Comment by tvoinarovskyi** [comment]
> Maybe add tests for errors? Removing non-existing user, adding already existing user?

Added as e2e tests, as those don't alter the state of the cluster. Please give it another view.

File: neuromation/cli/admin.py [hunk 6]
```diff
@@ -26,4 +28,47 @@ def admin() -> None:
...
+async def add_cluster_user(
+    root: Root, cluster_name: str, user_name: str, role: str
+) -> None:
+    """
+    Add user access to specified cluster with one of 3 roles: admin, manager or user
+    """
+    await root.client._admin.add_cluster_user(cluster_name, user_name, role)
+    if not root.quiet:
+        click.echo(
+            f"Added {click.style(user_name, bold=True)} to cluster "
```
Comment (serhiy-storchaka) [inline neuromation/cli/admin.py]: Maybe use attributes of the result of `add_cluster_user()`: `user.user_name`, `user.role`? Just for the case if the server changes them.

File: neuromation/api/admin.py [hunk 7]
```diff
@@ -38,6 +38,25 @@ def __init__(self, core: _Core, config: Config) -> None:
...
+            payload = await resp.json()
+            return _cluster_user_from_api(payload)
+
+    async def remove_cluster_user(self, cluster_name: str, user_name: str) -> None:
+        url = self._config.admin_url / "clusters" / cluster_name / "users" / user_name
+        auth = await self._config._api_auth()
+
+        async with self._core.request("DELETE", url, auth=auth) as resp:
+            # No content response
+            await resp.release()
```
Comment (asvetlov) [inline neuromation/api/admin.py]:
```suggestion
            pass
```
The response is released on exit from `async with`

File: neuromation/cli/admin.py [hunk 8]
```diff
@@ -26,4 +28,47 @@ def admin() -> None:
     )
 
 
+@command()
+@click.argument("cluster_name", required=True, type=str)
+@click.argument("user_name", required=True, type=str)
+@click.argument(
```
Comment (asvetlov) [inline neuromation/cli/admin.py]: That's interesting: `click` renders `ROLE` argument as mandatory according to README.

Comment (tvoinarovskyi) [reply neuromation/cli/admin.py]: Yea and if I remove metavar it prints bad error messages:

```
Invalid value for "[[admin|manager|user]]": invalid choice: my_role. (choose from admin, manager, user)
```

Don't know which is better

Comment (serhiy-storchaka) [reply neuromation/cli/admin.py]: What if set `metavar="[ROLE]"`?

Comment (tvoinarovskyi) [reply neuromation/cli/admin.py]: Yea, at least that looks better

**Comment by tvoinarovskyi** [comment]
@serhiy-storchaka Added test to check that we print values returned from the server, not from input params.

**Pull request merged**